### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by zemola**
- Create a class `Corrector` with a method `correct_name` that does the following:
- > Make sure the first letter of the given name is a capital letter.
- > Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.

- Set an instance of Corrector in Person on initialization.
- In Person add a method `validate_name` that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -20,4 +22,8 @@ def of_age?
   def can_use_services?
     of_age? || @parent_permission
   end
+
+  def validate_name
+    @name = @corrector.correct_name(name)
```
Comment (TedLivist) [inline person.rb]: - [ ] You should correct the `@name` (with `@`) instance variable (not `name`). Kindly fix this.

  You should have something like this `@name = @corrector.correct_name(@name)`

File: corrector.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      name.capitalize[0..9]
+    else
+      name.capitalize
+    end
+  end
+end
+
+def validate_name
+  @name = @corrector.correct_name(name)
+end
```
Comment (TedLivist) [inline corrector.rb]: - [ ] In keeping with the YAGNI (You Ain't Gonna Need It) rule, I would recommend you remove this method from this class, as you aren't (and won't be) using it.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @zemola,

Good job so far!
✔️ Good documentation
✔️ Implementation of Corrector class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @zemola, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**